src/proxy.cpp

```diff
@@ -129,12 +147,53 @@ int forward (
         rc = to_->send (&msg_, more ? ZMQ_SNDMORE : 0);
         if (unlikely (rc < 0))
             return -1;
+
         if (more == 0)
             break;
     }
+
+    // A multipart message counts as 1 packet:
+    from_stats->msg_in++;
+    from_stats->bytes_in+=complete_msg_size;
+    to_stats->msg_out++;
+    to_stats->bytes_out+=complete_msg_size;
+
+    return 0;
+}
+
+int reply_stats(
+        class zmq::socket_base_t *control_,
+        zmq_socket_stats_t* frontend_stats,
+        zmq_socket_stats_t* backend_stats)
+{
+    // first part: frontend stats
+
+    zmq::msg_t stats_msg1, stats_msg2;
+    int rc = stats_msg1.init_size( sizeof(zmq_socket_stats_t) );
+    if (unlikely (rc < 0))
+        return close_and_return (&stats_msg1, -1);
+
+    memcpy( stats_msg1.data(), (const void*) frontend_stats, sizeof(zmq_socket_stats_t) );
+
+    rc = control_->send (&stats_msg1, ZMQ_SNDMORE);
+    if (unlikely (rc < 0))
+        return close_and_return (&stats_msg1, -1);
+
+    // second part: backend stats
+
+    rc = stats_msg2.init_size( sizeof(zmq_socket_stats_t) );
+    if (unlikely (rc < 0))
+        return close_and_return (&stats_msg2, -1);
+    memcpy( stats_msg2.data(), (const void*) backend_stats,  sizeof(zmq_socket_stats_t) );
```

Comment:
There should be a space after the function name, but no spaces inside the braces